pkg/wharfapi/requestHandler.go

```diff
@@ -16,6 +16,91 @@ var redacted = "*REDACTED*"
 var tokenPatternJSON = regexp.MustCompile(`("token"\s*:\s*"([a-zA-Z\d\s]+)")\s*`)
 var tokenReplacementJSON = fmt.Sprintf(`"token":"%s"`, redacted)
 
+func newRequest(method, authHeader, baseURL, path string, q url.Values, body []byte) (*http.Request, error) {
+	u, err := newURL(baseURL, path, q)
+	if err != nil {
+		return nil, err
+	}
+	return newRequestFromURL(method, authHeader, u, body)
+}
+
+func newURL(baseURL, path string, q url.Values) (*url.URL, error) {
+	u, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, err
+	}
+	u.Path = path
+	u.RawQuery = q.Encode()
+	return u, nil
+}
+
+func newRequestFromURL(method, authHeader string, u *url.URL, body []byte) (*http.Request, error) {
+	urlStr := u.String()
+	req, err := http.NewRequest(method, urlStr, bytes.NewReader(body))
+	if err != nil {
+		log.Error().WithError(err).Message("Failed preparing HTTP request.")
+		return nil, err
+	}
+
+	if authHeader != "" {
+		req.Header.Add("Authorization", authHeader)
+	}
+
+	return req, nil
+}
+
+func doRequest(req *http.Request) ([]byte, error) {
+	client := &http.Client{}
+	response, err := client.Do(req)
+
+	var redactedURL = redactTokenInURL(req.URL.String())
+	var withRequestMeta = func(ev logger.Event) logger.Event {
+		return ev.
+			WithString("method", req.Method).
+			WithString("url", redactedURL)
+	}
+	withRequestMeta(log.Debug()).Message("")
```

Comment:
since wharf-core v1.3.0 we got `.WithFunc`. Changing to that might increase readability